### DISCUSSION TITLE
Unable to add quotes for expression column generation.

### DISCUSSION THREAD

**Discussion by Tanmayshetty** [saulpw/visidata#2430] [Q&A] [answered]
I was trying to follow this approach to generate a concatenated column https://www.visidata.org/docs/columns/#derived

![image](https://github.com/saulpw/visidata/assets/8419309/caf0cee2-c6ff-4dc2-b90d-4ae755472c89)

But I am not able to enter quotes in the terminal on Windows.
If I copy and paste the expression 
> Year + '-' + Month + '-' + Day

I get the below
![image](https://github.com/saulpw/visidata/assets/8419309/1b020a80-7be6-4b42-8a71-07d9bfea659c)

I have checked this with wezterm and Windows terminal.

Am I missing something here?

**Comment by saulpw** [comment]
Hi @Tanmayshetty, what version of VisiData are you using?

**Reply by Tanmayshetty** [reply]
@saulpw I am using v3.0.2

> saul.pw/VisiData v3.0.2

**Reply by saulpw** [reply]
Can you try with the most recent version on develop?

**Reply by Tanmayshetty** [reply]
With the dev version I get the same issue

>  visidata --version
>  saul.pw/VisiData v3.1dev
> 
> python --version
> Python 3.12.4

Attached a video 

https://github.com/saulpw/visidata/assets/8419309/6df901f8-fb21-4283-bf08-dbb7fa0bbc98

**Comment by midichef** [comment]
I can reproduce the lack of `'` on Windows.
setup:  Windows Server 2022 Datacenter, version 21H2, running visidata (2.9, 2.11.1, 3.0, 3.0.2, 3.1dev) in Powershell. Keyboard setting in Windows `Settings`->`Language`->`Keyboard` is `English (United States)`.

Other keys are affected too. I get no response from the keys `'` and `]`, nor from pressing them with `Shift` held: `"` and `}`. @Tanmayshetty do these other three keys (`]`, `"`, and `}`), have any effects for you in Visidata?

On my system:
For `'` the statusbar displays `SHF_PADENTER`.
For `"` the statusbar displays `CTL_PADENTER`.
For `]` and `}` the statusbar displays `Ctrl+Space` in 3.x; in v2.11.1 it shows `Ctrl+@`.

Other keys work as expected. All keys visible on a standard US English keyboard, work as expected. They also work when combined with `Shift`. Alphabet keys, with `Ctrl`, also work. So do arrow keys, `Home`/`End`, `Delete`. Function keys also work, `F1` thru `F12`, and all twelve combined with `Shift`.  With `Ctrl`, `F2` through `F12` give responses, though `Ctrl`+`F1` does not seem to do anything.

On this same Windows Server system, visidata works fine if I run it in Ubuntu inside WSL (Windows Subsystem for Linux).

Edited to add:  these four keys `'` etc, work fine inside Powershell, just not inside Visidata. If I try to bypass the keyboard by pasting from the system clipboard, when I paste `'`, I get the same result as if I press it on the keyboard: `SHF_PADENTER`.

**Reply by Tanmayshetty** [reply]
> Do these other three keys (], ", and }), have any effects for you in Visidata?

These keys do not work as well.

**Comment by midichef** [comment, marked as answer]
Okay, I now see that this bug has been discussed before in https://github.com/saulpw/visidata/issues/2119 and https://github.com/saulpw/visidata/issues/2062 and  #1841.

The last update to those threads was that `windows_curses` only works with a past version, 2.3.0. It was hoped then that 2.3.2 would fix it.

I can report that 2.3.2 does not fix it, nor does v2.3.3. I tested them on Python 3.12.4, where the versions of `windows_curses` available to me in pip are 2.3.2a1, 2.3.2, 2.3.3a1, and 2.3.3. None of these work satisfactorily. The first two (.2a1 and .2) cause visidata to exit on startup without showing any data screen; `visidata a.tsv` exits after saying `opening a.tsv as tsv`. That last two have the problem of incorrectly reading `'` and `"` and `]` and `}`.

I can report that downgrading `windows_curses` still works: `pip install windows_curses==2.3.0`. To use pip for this, I had to use Python 3.10, specifically 3.10.11, which is the last version of 3.10 that has a binary installer for Windows. However, that version is not an acceptable solution as it's missing several security fixes from 3.10.14. If you can figure out how to get Python 3.12.4 to use `windows_curses` 2.3.0, that may work.

But the solution I recommend is to run visidata inside Windows Subsystem for Linux.

**Reply by Tanmayshetty** [reply]
> I now see that this bug has been discussed before in https://github.com/saulpw/visidata/issues/2119 and https://github.com/saulpw/visidata/issues/2062.

Thanks for the issue reference.
I will use the workaround specified here.
